File: src/test/java/utils/XorTest.java
package utils;

import org.junit.Test;
import static org.junit.Assert.*;

public class XorTest {
    @Test
    public void sameLengthTest() {
        byte[] byte1 = Codec.fromHex("12345678");
        byte[] byte2 = Codec.fromHex("90abcdef");
        byte[] expected = Codec.fromHex("829f9b97");
        byte[] result = Xor.sameLength(byte1, byte2);
        assertArrayEquals(expected, result);
    }
}

File: src/test/kotlin/utils/byteops/XorTest.kt
package utils.byteops

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import utils.codec.Hex
import utils.codec.toBytes

@ExperimentalUnsignedTypes
class XorTest : FunSpec({
    test("Given test cases pass") {
        val byte1 = Hex("12345678").toBytes()
        val byte2 = Hex("90abcdef").toBytes()
        val expected = Hex("829f9b97").toBytes()
        val result = byte1 xor byte2
        result shouldBe expected
    }
})
